### PR TITLE
VIH-6657 Log_Data_related_to_users_connection_to_the_Hearing_Room

### PR DESCRIPTION
**Pull request by RocheL**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-6657
### Change description ###
 - added logging for camera, microphone change
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 1]
```diff
@@ -284,11 +290,11 @@ export abstract class WaitingRoomBaseComponent {
     }
 
     call() {
-        this.logger.info('[WR] - calling pexip');
+        this.logger.info(`${this.loggerPrefix} calling pexip`);
         const pexipNode = this.hearing.getConference().pexip_node_uri;
         const conferenceAlias = this.hearing.getConference().participant_uri;
         const displayName = this.participant.tiled_display_name;
-        this.logger.debug(`[WR] - Calling ${pexipNode} - ${conferenceAlias} as ${displayName}`);
+        this.logger.debug(`${this.loggerPrefix} Calling ${pexipNode} - ${conferenceAlias} as ${displayName}`);
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:297]: could you add the conference and participant as the third param like the other logs do please

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 2]
```diff
@@ -314,15 +320,15 @@ export abstract class WaitingRoomBaseComponent {
     }
 
     handleCallSetup(callSetup: CallSetup) {
-        this.logger.debug('[WR] - Conference has setup');
+        this.logger.debug(`${this.loggerPrefix} Conference has setup`);
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:323]: could you add the conference and participant as the third param like the other logs do please

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 3]
```diff
@@ -339,7 +345,9 @@ export abstract class WaitingRoomBaseComponent {
         this.errorCount++;
         this.connected = false;
         this.updateShowVideo();
-        this.logger.error(`[WR] - Error from pexip. Reason : ${error.reason}`, new Error(error.reason), { pexipError: error });
+        this.logger.error(`${this.loggerPrefix} Error from pexip. Reason : ${error.reason}`, new Error(error.reason), {
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:348]: could you add the conference and participant as the third param like the other logs do please

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
Could you make sure all the logs have at least conference and participant in the payload please. thanks

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts (hunk 1 above)
Comment (RocheL) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts]: added conference id and participant id

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts (hunk 2 above)
Comment (RocheL) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts]: added conference id and participant id

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts (hunk 3 above)
Comment (RocheL) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:360]: added conference id and participant id

**Review by shaed-parkar — COMMENTED** [review]
LGTM

**Review by shaed-parkar — APPROVED** [review]

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
You have reverted some test assertions in regards to handling pexip errors.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-waiting-room/tests/judge-waiting-room.component.spec.ts [hunk 4]
```diff
@@ -261,7 +261,11 @@ describe('JudgeWaitingRoomComponent when conference exists', () => {
         expect(component.heartbeat.kill).toHaveBeenCalled();
         expect(component.errorCount).toBeGreaterThan(currentErrorCount);
         expect(component.showVideo).toBeFalsy();
-        expect(errorService.handlePexipError).toHaveBeenCalledWith(payload);
+        expect(errorService.goToServiceError).toHaveBeenCalledWith(
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-waiting-room/tests/judge-waiting-room.component.spec.ts:264]: Please ensure the handlePexipError is called here

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/tests/participant-waiting-room.component.videocall.events.spec.ts [hunk 5]
```diff
@@ -196,7 +196,11 @@ describe('ParticipantWaitingRoomComponent video call events', () => {
         expect(component.heartbeat.kill).toHaveBeenCalled();
         expect(component.errorCount).toBeGreaterThan(currentErrorCount);
         expect(component.showVideo).toBeFalsy();
-        expect(errorService.handlePexipError).toHaveBeenCalledWith(payload);
+        expect(errorService.goToServiceError).toHaveBeenCalledWith(
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/tests/participant-waiting-room.component.videocall.events.spec.ts:199]: Please ensure the handlePexipError is called here

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 6]
```diff
@@ -339,15 +353,27 @@ export abstract class WaitingRoomBaseComponent {
...
+            pexipError: error,
+            conference: this.conference.id,
+            participant: this.participant.id
+        });
+        if (error.reason.includes('Error connecting to conference')) {
+            this.errorService.goToServiceError('Your connection was lost');
+            return;
+        }
+        this.errorService.goToServiceError(
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:365]: Please ensure the handlePexipError is called here

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
few more oddities from the merge

**Review by shaed-parkar — APPROVED** [review]
Nice work!

**Pull request merged**